Assets/HoloToolkit/Utilities/Scripts/BoundingBoxHelper.cs

```diff
@@ -0,0 +1,327 @@
+    /// <returns>a vector representing the bottom most edge center of the face</returns>
+    public Vector3 GetFaceBottomCentroid(int index)
+    {
+        Vector3[] edgeCentroids = GetFaceEdgeMidpoints(index);
+
+        Vector3 leastYPoint = edgeCentroids[0];
+        for (int i = 1; i < 4; ++i)
+        {
+            leastYPoint = edgeCentroids[i].y < leastYPoint.y ? edgeCentroids[i] : leastYPoint;
+        }
+        return leastYPoint;
+    }
+
+    /// <summary>
+    /// This function returns the four couners of a face of a bounding cube specified by index.
+    /// </summary>
+    /// <param name="index">the index of the face of the bounding cube. 0-5</param>
+    /// <returns>an array of 4 vectors</returns>
+    public Vector3[] GetFaceCorners(int index)
+    {
+        int[] faceIndices = GetFaceIndices(index);
+
+        if (faceIndices.Length == 4)
+        {
+            Vector3[] face = new Vector3[4];
+            face[0] = worldBoundingCorners[faceIndices[0]];
+            face[1] = worldBoundingCorners[faceIndices[1]];
+            face[2] = worldBoundingCorners[faceIndices[2]];
+            face[3] = worldBoundingCorners[faceIndices[3]];
+            return face;
+        }
+
+        return new Vector3[0];
+    }
+
+    /// <summary>
+    /// This function gets the index of the face of the bounding cube that is most facing the lookAtPoint.
+    /// This could be the headPosition or camera position if the face that was facing the view is desired.
+    /// </summary>
+    /// <param name="lookAtPoint">the world coordinate to test which face is desired</param>
+    /// <returns>an integer representing the index of the bounding box faces</returns>
+    public int GetIndexOfForwardFace(Vector3 lookAtPoint)
+    {
+        int highestDotIndex = -1;
+        float hightestDotValue = float.MinValue;
+        for (int i = 0; i < 6; ++i)
+        {
+            Vector3 a = (lookAtPoint - GetFaceCentroid(i) ).normalized;
+            Vector3 b = GetFaceNormal(i);
+            float dot = Vector3.Dot(a, b);
+            if (hightestDotValue < dot)
+            {
+                hightestDotValue = dot;
+                highestDotIndex = i;
+            }
+        }
+        return highestDotIndex;
+    }
+
+    /// <summary>
+    /// This is the static function to call to get the non-Axis-aligned bounding box corners one time only.
+    /// Use this function if the calling object only needs the info once. To get an updated boundingbox use the
+    /// function above---UpdateNonAABoundingBoxCornerPositions(...);
+    /// </summary>
+    /// <param name="target">The gameObject whose bounding box is desired</param>
+    /// <param name="boundsPoints">the array of 8 points that will be filled</param>
+    /// <param name="ignoreLayers">a LayerMask variable</param>
+    public static void GetNonAABoundingBoxCornerPositions(GameObject target, List<Vector3> boundsPoints, LayerMask ignoreLayers)
```

Comment:
It seems like we have a lot of duplicated code that does almost identical things.

Could consolidate these?

Reply:
yes- great idea! both the main functions can share a new static function that does the bulk of the calculations.